base/uk.ac.stfc.isis.ibex.configserver/src/uk/ac/stfc/isis/ibex/configserver/editing/EditableConfiguration.java

```diff
@@ -750,6 +750,26 @@ public List<String> getGroupNames() {
 		.map(Group::getName)
 		.collect(Collectors.toCollection(ArrayList::new));
     }
+    
+    /**
+     * Get the current list of groups in the configuration.
+     * Excluding groups from components.
+     * 
+     * @return the list of configuration group names
+     */
+    public List<String> getConfigGroupNames() {
+    	var allGroups = getGroupNames();
+    	
+    	var componentGroups = new ArrayList<String>();
+    	for (Configuration comp : editableComponents.getSelected()) {
+    		for (Group group : comp.getGroups()) {
+    			componentGroups.add(group.getName());
+    		}
+    	}
+    	
+    	allGroups.removeAll(componentGroups);
```

Comment:
If I have the same group name defined in both a config and a component then it looks like this will no longer let me add it to the config group - which should be allowed, I think? 

This is a valid workflow and the GUI should merge groups with the same name from the config & components. As a side note this seems buggy in the GUI at the moment - if I look at EMU which has this kind of setup and go to their groups menu, clicking on any group causes the GUI to complain "group names must be unique". So you may need to fix that bug first before you can test the workflow in this comment...